lua/lualine/init.lua

```diff
@@ -279,6 +279,9 @@ local function setup_theme()
       }, true, {})
       theme = require 'lualine.themes.gruvbox'
     end
+  else
+    -- use the provided theme as-is, assuming it's a dictionary
```

Comment:
no need to assume anything, better to have `elseif type == table` here

and below `else` with error log + defaulting to gruvbox

Reply:
Merge this one for now. It will fix the bug on master . I'll make a pr regarding the changes you're looking for.

Reply:
Sorry, was working during the day so wasn't able to update it. I saw there's another PR opened for this so I'll resolve this convo. Thanks for the merge.

Reply:
np, thanks for noticing, I've applied additional changes already